### ISSUE TITLE
Support weekly summaries

### ISSUE THREAD

**Issue by christopherjbaker** [m-thompson-code/opensource-analytics-dashboard#2]
When you look at the data over a large period of time, such as a quarter, it highlights a weird issue inherent in the data: packages are used more on weekdays than weekends. It makes it difficult to see patterns when the lines move so erratically. If the points were weekly instead of daily, this would smooth out and be more visually meaningful.

I wouldn't make it always do this though, I'd make it an option. For long-running projects, having monthly or quarterly breakdowns might be useful too (though those get more complicated since it's not a consistent number of days). I did this on another project and when the date changed, we defaulted to a scale based on the size range (targeting a specific number of points in the graph), but that's entirely optional too.

<img width="1086" alt="image" src="https://user-images.githubusercontent.com/980338/184028859-95e7c2a6-75c8-4073-b6ba-da9612eb9a83.png">

**Comment by m-thompson-code** [comment]
I agree with this. It shouldn't be too much work to add these features since it's mostly messing with the data than it being ui stuff
